status.go

```diff
@@ -118,3 +118,16 @@ func (r *StatusResp) WriteTo(w *Writer) error {
 
 	return w.writeCrlf()
 }
+
+// ErrStatusResp can be returned by a Handler to replace the default status
+// response. The response tag must be empty.
+//
+// To suppress default response, Resp should be set to nil.
+type ErrStatusResp struct {
+	// Response to send instead of default.
+	Resp *StatusResp
+}
+
+func (err *ErrStatusResp) Error() string {
+	return ""
```

Comment:
Unrelated to this PR, but this is a little bit weird. Maybe we should change this.

Reply:
I am not entirely sure about contexts where StatusResp is used.
Would this work fine?
```go
func (err *ErrStatusResp) Error() string {
	if err.Resp == nil {
		return "imap: suppressed response"
	}
	return err.Resp.Info
}
```

Reply:
Yeah, this looks good to me. It's just to avoid surprising and hard-to-debug error messages.